api/waffle/serializers.py

```diff
@@ -0,0 +1,23 @@
+from rest_framework import serializers as ser
+
+from api.base.serializers import JSONAPISerializer, TypeField
+from api.base.waffle_decorators import waffle_feature_is_active
+
+class WaffleSerializer(JSONAPISerializer):
+    id = ser.SerializerMethodField()
+    type = TypeField()
+    name = ser.CharField(required=True, help_text='The name of the waffle object')
+    active = ser.SerializerMethodField()
+    note = ser.CharField(required=False, allow_blank=True, help_text='Describe where the waffle object is used.')
+
+    def get_active(self, obj):
+        """
+        Use waffle_feature_is_active helper to determine if waffle flag, sample, or switch is active
+        """
+        return waffle_feature_is_active(self.context.get('request'), type(obj).__name__, obj.name)
+
+    def get_id(self, obj):
+        return '{}_{}'.format(type(obj).__name__.lower(), obj.id)
+
+    class Meta:
+        type_ = 'waffle'
```

Comment:
Would it make sense to have different types for switches, samples, and flags (e.g. `waffle_flags`). That would allow the frontend to distinguish between the types.

As @brianjgeiger pointed out in https://github.com/CenterForOpenScience/osf.io/pull/7734#issuecomment-347988818 , it's ok to have an endpoint return different types.

Reply:
Yes, I can return different types.  (Side note: I am also returning the type in the `IDField`.  Samples, switches, and flags are separate models, so both a sample and a switch can have an ID of 1.  If I am fetching both a sample and a switch in an Ember app with id=1, Ember will not like the duplicate ID's.  So I am returning `flag-1` and `sample-1` as the id's.

Also, Ember-side, we will need to create a flag model, switch model, and sample model, instead of just a single waffle model.